clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -1025,59 +1013,63 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
+      llvm::APSInt ArgVal = CE->getResultAsAPSInt();
+      addSYCLIInterval(ArgVal.getSExtValue());
+    }
 
     if (const auto *IntelFPGAMaxConcurrency =
             dyn_cast<SYCLIntelFPGAMaxConcurrencyAttr>(A)) {
-      setSYCLMaxConcurrencyEnable();
-      setSYCLMaxConcurrencyNThreads(IntelFPGAMaxConcurrency->getNThreadsExpr()
-                                        ->getIntegerConstantExpr(Ctx)
-                                        ->getSExtValue());
+      const auto *CE =
+          cast<ConstantExpr>(IntelFPGAMaxConcurrency->getNThreadsExpr());
+      llvm::APSInt ArgVal = CE->getResultAsAPSInt();
+      addSYCLMaxConcurrencyNThreads("llvm.loop.max_concurrency.count",
+                                    ArgVal.getSExtValue());
     }
 
     if (const auto *IntelFPGALoopCountAvg =
             dyn_cast<SYCLIntelFPGALoopCountAttr>(A)) {
-      unsigned int Count = IntelFPGALoopCountAvg->getNTripCount()
-                               ->getIntegerConstantExpr(Ctx)
-                               ->getSExtValue();
+      const auto *CE =
+          cast<ConstantExpr>(IntelFPGALoopCountAvg->getNTripCount());
+      llvm::APSInt ArgVal = CE->getResultAsAPSInt();
       const char *Var = IntelFPGALoopCountAvg->isMax()
                             ? "llvm.loop.intel.loopcount_max"
                             : IntelFPGALoopCountAvg->isMin()
                                   ? "llvm.loop.intel.loopcount_min"
                                   : "llvm.loop.intel.loopcount_avg";
-      setSYCLIntelFPGAVariantCount(Var, Count);
+      addSYCLIntelFPGAVariantCount(Var, ArgVal.getSExtValue());
     }
 
     if (const auto *IntelFPGALoopCoalesce =
             dyn_cast<SYCLIntelFPGALoopCoalesceAttr>(A)) {
-      if (auto *LCE = IntelFPGALoopCoalesce->getNExpr())
-        setSYCLLoopCoalesceNLevels(
-            LCE->getIntegerConstantExpr(Ctx)->getSExtValue());
-      else
-        setSYCLLoopCoalesceEnable();
+      if (const auto *LCE = IntelFPGALoopCoalesce->getNExpr()) {
+        const auto *CE = cast<ConstantExpr>(LCE);
+        llvm::APSInt ArgVal = CE->getResultAsAPSInt();
+        addSYCLLoopCoalesceNLevels(ArgVal.getSExtValue());
+      } else {
+        addSYCLLoopCoalesceEnable();
+      }
     }
 
     if (isa<SYCLIntelFPGADisableLoopPipeliningAttr>(A))
       setSYCLLoopPipeliningDisable();
 
     if (const auto *IntelFPGAMaxInterleaving =
             dyn_cast<SYCLIntelFPGAMaxInterleavingAttr>(A)) {
-      setSYCLMaxInterleavingEnable();
-      setSYCLMaxInterleavingNInvocations(IntelFPGAMaxInterleaving->getNExpr()
-                                             ->getIntegerConstantExpr(Ctx)
-                                             ->getSExtValue());
+      const auto *CE = cast<ConstantExpr>(IntelFPGAMaxInterleaving->getNExpr());
+      llvm::APSInt ArgVal = CE->getResultAsAPSInt();
+      addSYCLMaxInterleavingNInvocations("llvm.loop.max_interleaving.count",
```

Comment:
Sorry, I was a bit unclear before -- I meant, rather than storing this information by passing the same string literal in here, use the string literal directly when iterating over `for (const auto &MI : Attrs.SYCLMaxInterleavingNInvocations) {` rather than using `MI.first` to hold that data temporarily.